lib/pacemaker/pcmk_sched_probes.c

```diff
@@ -251,24 +251,24 @@ pcmk__probe_rsc_on_node(pe_resource_t *rsc, pe_node_t *node)
     // We've eliminated all cases where a probe is not needed, so now it is
     probe = probe_action(rsc, node);
 
-    /* Order the probe relative to the parent -- or the resource itself if
-     * cloned or a fence device when unfencing is used.
+    /* Below, we will order the probe relative to start or reload. If this is a
+     * clone instance, the start or reload is for the entire clone rather than
+     * just the instance. Otherwise, the start or reload is for the resource
+     * itself.
      */
-    if ((pcmk_is_set(rsc->flags, pe_rsc_fence_device)
-         && pcmk_is_set(rsc->cluster->flags, pe_flag_enable_unfencing))
-        || !pe_rsc_is_clone(top)) {
+    if (!pe_rsc_is_clone(top)) {
         top = rsc;
     }
 
+    /* Prevent a start if the resource can't be probed, but don't cause the
+     * resource or entire clone to stop if already active.
+     */
     if (!pcmk_is_set(probe->flags, pe_action_runnable)
```

Comment:
Isn't `!pcmk_is_set(probe->flags, pe_action_runnable)` already implied by `pe_order_runnable_left`?

Reply:
That does appear to be redundant. It was added in faf44d811.

There could be a difference in behavior if runnable is changed between this point and when actions are updated, but it's unclear which behavior would be correct. Removing the check gives this change in one test:

    -  * Start      galera-bundle-0            (         metal-1 )  due to unrunnable galera-bundle-docker-0 start (blocked)
    +  * Start      galera-bundle-0            (         metal-1 )  blocked

which is odd. It looks like setting pe_order_runnable_left when first is runnable can still have some effect, which sounds brittle. I didn't nail it down, but it looks like with the original code, the start gets blocked when processing the "first" action, and with dropping the runnable check here, the start gets blocked when processing the "then" action. It hits the same code though, so I'm not sure why there's no reason given.

I'm inclined to leave it as-is without spending a lot more time investigating it.